src/blueapi/service/main.py

```diff
@@ -38,19 +41,22 @@ async def get_devices(handler: Handler = Depends(get_handler)):
 
 @app.get("/device/{name}", response_model=DeviceModel)
 async def get_device_by_name(name: str, handler: Handler = Depends(get_handler)):
+    """Retrieve information about a devices by its (unique) name."""
     try:
         return DeviceModel.from_device(handler.context.devices[name])
     except KeyError:
         raise HTTPException(status_code=404, detail="Item not found")
 
 
-@app.put("/task/{name}")
-async def execute_task(
+@app.put("/task/{name}", response_model=TaskResponse)
+async def submit_task(
     name: str,
     task: Mapping[str, Any] = Body(..., example={"detectors": ["x"]}),
     handler: Handler = Depends(get_handler),
 ):
+    """Submit a task onto the worker queue."""
     handler.worker.submit_task(name, RunPlan(name=name, params=task))
+    return TaskResponse(task_name=f"Task {name} submitted")
```

Comment:
This needs to literally be the same name, it is meant to be computer-readable, not human-readable.
```suggestion
    return TaskResponse(task_name=name)
```